benchmarks/timeseries/serializable_ringbuffer.py

```diff
@@ -0,0 +1,82 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Benchmarks the `SerializableRingbuffer` class."""
+
+from __future__ import annotations
+
+import fnmatch
+import os
+import time
+from datetime import datetime, timedelta
+from typing import Any
+
+import numpy as np
+
+from frequenz.sdk.timeseries import Sample
+from frequenz.sdk.timeseries._serializable_ringbuffer import SerializableRingBuffer
+
+FILE_NAME = "ringbuffer.pkl"
+
+FIVE_MINUTES = timedelta(minutes=5)
+
+
+def delete_files_with_prefix(prefix: str) -> None:
+    """Delete all files starting with the given prefix.
+
+    Args:
+        prefix: Prefix of the files to delete
+    """
+    for file in os.listdir():
+        if fnmatch.fnmatch(file, prefix + "*"):
+            os.remove(file)
+
+
+def benchmark_serialization(
+    ringbuffer: SerializableRingBuffer[Any], iterations: int
+) -> float:
+    """Benchmark the given buffer `iteration` times.
+
+    Args:
+        ringbuffer: Ringbuffer to benchmark to serialize.
+        iterations: amount of iterations to run.
+    """
+    total = 0.0
+    for _ in range(iterations):
+        start = time.time()
+        ringbuffer.dump()
+        SerializableRingBuffer.load(FILE_NAME)
+        end = time.time()
+        total += end - start
+        delete_files_with_prefix(FILE_NAME)
+
+    return total / iterations
+
+
+SIZE = 4000_000
+ITERATIONS = 100
+
+
+def main() -> None:
+    """Run Benchmark."""
+    ringbuffer = SerializableRingBuffer(
+        np.arange(0, SIZE, dtype=np.float64), timedelta(minutes=5), FILE_NAME
+    )
+
+    print("size:", SIZE)
+    print("iterations:", ITERATIONS)
+
+    for i in range(0, SIZE, 10000):
+        ringbuffer.update(
+            Sample(datetime.fromtimestamp(200 + i * FIVE_MINUTES.total_seconds()), i)
+        )
+
+    print(
+        "Avg time for Pickle dump/load:  ",
+        benchmark_serialization(ringbuffer, ITERATIONS),
+        "s",
+    )
```

Comment:
why not a format-string?

Reply:
Does it make much difference? I think this part was written by GPT ;)

Reply:
Slap GPT until it uses f-strings.

Reply:
also good reason to rewrite,  to keep all the gpt code out.

Reply:
I disagree. If GPT generates specific code to my instructions with context I have given then I don't see any danger of any copyright related code, certainly not for snippets like this and of this size.

Reply:
And

```
    print(
        "Avg time for Pickle dump/load:  "
        f"{benchmark_serialization(ringbuffer, ITERATIONS)}s"
    )
```

Does not look specifically more readable to me...

Reply:
But IMHO:
```py
elapsed_secs = benchmark_serialization(ringbuffer, ITERATIONS)
print(f"Avg time for Pickle dump/load: {elapsed_secs}s")
```
Does look more readable IMHO. That said, I don't think this minor formatting issue is worth the time of the discussion (except for the more general GPT part maybe). If you are very convinced this is good enough, leave it as is.

About if to include or not code produced by GPT I won't give an opinion because it is a complex topic that is ongoing and nobody has a clear answer to that question yet. To be on the safe side I would also try to keep non-trivial code generated by GPT out.